advent_of_code/day2/gift.py

```diff
@@ -0,0 +1,39 @@
+class Gift:
+	def __init__(self, dimensions):
+		self.dimensions = dimensions
+
+	def calculation_of_wrapping_paper(self):
+		total = 0
+
+		for dimension in self.dimensions:
+			try:
+				length, width, height = self.get_sizes_of(dimension)
+			except Exception:
+				raise Exception('Missing parameters')
+
+			total += self.amount_of_paper(length, width, height)
+
+		return total
+
+	def get_sizes_of(self, dimension):
```

Comment:
Acho que esse método poderia ter um underscore no início, porque ele seria um método "privado". Compreendes?

Então `_get_sizes_of`.

Reply:
Acho que não, objetivo de deixar o metódo privado e não deixa eu acessar ele fora da classe?

Reply:
Sim, o objetivo do método privado é não deixar acessar ele fora da classe. Só a própria classe vai ter acesso a ele.

Mas como já foi dito umas 3 ou 4 vezes lá no grupo: no Python não existem métodos privados. Então para "dizer que" um método é privado, é uma prática comum colocar um underscore `_` antes do nome do método.
Então com isso só assumiremos que esse método não pode ser chamado de fora.
E talvez uma coisa legal de não ter o método privado no Python é que você pode fazer um teste unitário pra cada um desses métodos. :)

Tipo, o exercício que vc está fazendo deixa bem claro qual é a interface pública que sua classe precisa, que é:
- Dizer quanto de `wrapping paper` é necessário.
- Dizer quanto de `feet ribbon` é necessário.

Né?

Ou seja, na interface pública da sua classe nós precisamos ter apenas 2 métodos, que no caso foram os que você colocou: `calculation_of_wrapping_paper` e `calculation_of_feet_ribbon ` e só!

Os outros métodos além desses são só pra melhorar a legibilidade e tal da sua classe... Que daí seriam os métodos privados...
Tanto que seu código estava funcionando já antes de vc criar esses 2 métodos ali... Vc só criou ele para seu código ficar mais bonito!

Conseguiu ver que não tem a necessidade da gente ter um método público de `get_sizes_of` e nem de `amount_of_paper`? Esses são métodos que só nossa própria classe vai usar para poder fazer seus cálculos dos 2 métodos públicos? :)

Concorda comigo até aqui? Consegui ser claro? Captou?